frontend/javascripts/oxalis/model/sagas/annotation_saga.ts

```diff
@@ -159,4 +178,75 @@ export function* watchAnnotationAsync(): Saga<void> {
   );
   yield* takeLatest("EDIT_ANNOTATION_LAYER", pushAnnotationLayerUpdateAsync);
 }
-export default [warnAboutSegmentationZoom, watchAnnotationAsync];
+
+export function* acquireAnnotationMutexMaybe(): Saga<void> {
+  yield* take("WK_READY");
+  const allowUpdate = yield* select((state) => state.tracing.restrictions.allowUpdate);
+  const annotationId = yield* select((storeState) => storeState.tracing.annotationId);
+  if (!allowUpdate) {
+    return;
+  }
+  const othersMayEdit = yield* select((state) => state.tracing.othersMayEdit);
+  const ONE_MINUTE = 1000 * 60;
+  let isInitialRequest = true;
+  let doesHaveMutex = false;
+  let shallTryAcquireMutex = othersMayEdit;
+
+  function onMutexStateChanged(canEdit: boolean, blockedByUser: APIUserCompact | null | undefined) {
+    if (canEdit) {
+      Toast.close("MutexCouldNotBeAcquired");
+      if (!isInitialRequest) {
+        Toast.success(messages["annotation.acquiringMutexSucceeded"]);
+        location.reload();
```

Comment:
I don't think that this should happen automatically. I'd add a button to the toast so that the users workflow isn't interrupted (maybe they were just inspecting things).

Reply:
Would it help if I added a get route for the mutex that only yields info about its current state, without trying to acquire it?

Reply:
I don't think so. At that point, the user already has acquired the mutex and the frontend will try to keep it. I don't see where such an info route would be useful 🤔